flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/casting/CastRulesTest.java

```diff
@@ -563,11 +570,16 @@
                         .fromCase(
                                 MAP(STRING().nullable(), INTERVAL(MONTH()).nullable()),
                                 mapData(entry(null, -123), entry(fromString("b"), null)),
-                                fromString("{null=-10-03, b=null}"))
+                                fromString("{NULL=-10-03, b=NULL}"))
+                        .fromCase(
+                                MAP(STRING().nullable(), INTERVAL(MONTH()).nullable()),
+                                mapData(entry(null, null)),
+                                fromString("{NULL=NULL}"))
```

Comment:
nit: Maybe adding the `false` as last argument makes it more visible, since otherwise the test cases are the same. This is done in other cases already, so it would be nice to have consistency.

Instead though, we could make the `fromCase` with the boolean arg private , and introduce a `fromCaseLegacy` to be even more visible. WDYT?

Reply:
Yeah actually I prefer `fromCaseLegacy` rather than the boolean arg, i'll do this change now